src/super_gradients/common/plugins/deci_client.py

```diff
@@ -1,24 +1,42 @@
-import boto3
 import hydra
 from hydra.core.global_hydra import GlobalHydra
 from omegaconf import DictConfig
 
-class MockClient:
+try:
+    from deci_lab_client.client import DeciPlatformClient
+except (ImportError, NameError):
+    from typing import Any as DeciPlatformClient
 
-    def __init__(self):
-        self.bucket_name = 'deci-model-repository-research'
-        self.prefix = 'mock_client'
-        self.s3 = boto3.client('s3')
+
+class DeciClient:
+    def __init__(self, lab_client: DeciPlatformClient):
```

Comment:
why not instantiate the DeciPlatformClient inside? why do we need to pass it all the way from outside?

Reply:
The lab client needs to be logged in for anything to work.
With `deci-adk`, we provide colabs and snippets that initialize the client and log in with it first, and then pass it to the class to fetch the model using it

Reply:
but login can happen on `DeciPlatformClient.__init__()`. it's not like we expect the user to pass a user name or token in the code. these are taken from env_variable or local file (like with was cli)

Reply:
The snippets we currently have do expect the user to pass a token in the code, though

Reply:
snippets are ok, but real code does not include tokens.
you can have an optional parameter `DeciPlatformClient.__init__(self, user: Optional[str], token: Optional[str])` but the DeciPlatformClient needs to know how to init without it (like wandb, was and other clients).
either take the token from env/file or prompt for login

Reply:
👍